### PR TITLE
[Coverity] Fix AUTO plugin issues

### PR DESCRIPTION
**Pull request by yangwang201911**
### Details:
 - *item1*
 - *...*

### Tickets:
 - 145032

### CODE REVIEW

**Comment by riverlijunjie** [comment]
Why not resolve this issue as ticket description?

image

**Comment by riverlijunjie** [comment]
and how about?
image

**Review by zhaixuejun1993 — APPROVED** [review]
LGTM

**Comment by yangwang201911** [comment]
> and how about? image

resolve this issue with replacing `std::unique_lock` with `std::lock_guard`

File: src/plugins/auto/src/auto_schedule.cpp [hunk 1]
```diff
@@ -456,32 +456,32 @@ bool AutoSchedule::schedule_to_worker_infer_request(ov::threading::Task pipeline
...
+            // if the device needed by customer is not ready, need to wait for it
+            wait_actual_compiled_model_ready();
+            devices.push_back(m_compile_context[ACTUALDEVICE].m_device_info);
+            if (!deviceChecker().check_if_device_in_list<DeviceInformation>(preferred_device, devices)) {
+                OPENVINO_THROW("The preferred device should be the selected device");
+            }
         } else {
-            if (m_compile_context[ACTUALDEVICE].m_is_already) {
-                devices.push_back(m_compile_context[ACTUALDEVICE].m_device_info);
+            // _acceleratorDevice could be the same as _cpuDevice, such as AUTO:CPU
```
Comment (riverlijunjie) [inline src/plugins/auto/src/auto_schedule.cpp:469]: Does this block not need lock protection?

Comment (yangwang201911) [reply src/plugins/auto/src/auto_schedule.cpp:469]: this block has been in the scope of the lock protection here.

**Review by riverlijunjie — APPROVED** [review]

**Pull request merged**